app/src/main/java/com/github/libretube/ui/fragments/PlayerFragment.kt

```diff
@@ -1040,6 +1036,119 @@ class PlayerFragment : BaseFragment(), OnlinePlayerOptions {
         }
     }
 
+    private fun setupDescription(
+        descTextView: TextView,
+        description: String,
+        exoPlayer: ExoPlayer
+    ) {
+        // detect whether the description is html formatted
+        if (description.contains("<") && description.contains(">")) {
+            descTextView.movementMethod = LinkMovementMethod.getInstance()
+            descTextView.text = HtmlParser.createSpannedText(
+                description,
+                LinkHandler { link ->
+                    // check if the link is a youtube link
+                    if (link.contains("youtube.com") || link.contains("youtu.be")) {
+                        // check if the link is a video link
+                        val videoId = getVideoIdIfVideoLink(link)
+                        if (!videoId.isNullOrEmpty()) {
```

Comment:
Instead of using lots of if conditions here, we should create a separate function which for example directly opens a new link and returns when the `videoId` is null or empty. That could also be partly applied for the following if conditions, just to keep the code easier to understand.

Reply:
I did this because the link could also be a channel or playlist link. In future we can open channel or playlist within the app instead of opening an external intent.